pkg/util/test/assertions/events.go

```diff
@@ -105,3 +108,121 @@ func ShouldResembleEventDefinitionDataClosures(actual interface{}, expected ...i
 	}
 	return success
 }
+
+var eventTimeout = test.Delay << 7
+
+func receiveEvent(v interface{}) (events.Event, string) {
+	switch ch := v.(type) {
+	case <-chan events.Event:
+		select {
+		case <-time.After(eventTimeout):
+			return nil, fmt.Sprintf("Timed out while waiting for event to arrive")
+		case ev := <-ch:
+			return ev, success
+		}
+	case <-chan test.EventPubSubPublishRequest:
+		select {
+		case <-time.After(eventTimeout):
+			return nil, fmt.Sprintf("Timed out while waiting for event publish request to arrive")
+		case req := <-ch:
+			select {
+			case <-time.After(eventTimeout):
+				return nil, fmt.Sprintf("Timed out while waiting for event publish response to be processed")
+			case req.Response <- struct{}{}:
+			}
+			return req.Event, success
+		}
+	}
+	return nil, fmt.Sprintf(needEventChannelCompatible, v)
+}
+
+func ShouldReceiveEventFunc(actual interface{}, expected ...interface{}) string {
```

Comment:
This could also use some godoc. Without reading the code, developers have zero information on what to pass here.

Reply:
Also for the funcs below.

Reply:
The expected usage is via `should.ReceieveEventFunc`, but will add